### PR TITLE
disposal == 2

### PR DESCRIPTION
**Pull request by NoamCohen48**
added case if disposal is 2
just like if disposal is 3, inserted a fake frame filled with background color. 

noam tz: for the encode part we change little things there and then we will try to change and talk with u

### CODE REVIEW

File: decoder.py [hunk 1]
```diff
@@ -90,6 +94,26 @@ def decode_logical_screen_descriptor(gif_stream: BitStreamReader, gif_object: Gi
...
 
+def background_frame(gif_object: Gif):
+    prev_image = gif_object.images[-1]
+    current_image = Image()
+    current_image.index_graphic_control_ex = None
+
+    current_image.left = prev_image.left
+    current_image.top = prev_image.top
+    current_image.width = prev_image.width
+    current_image.height = prev_image.height
```
Comment (noamTzuber) [inline decoder.py:105]: r u sure we not need the full size of the gif?

File: decoder.py [hunk 2]
```diff
@@ -55,12 +55,16 @@ def decode_gif(io: typing.BinaryIO) -> Gif:
...
 
-            if gif_object.graphic_control_extensions[
-                gif_object.images[LAST_ELEMENT].index_graphic_control_ex].disposal == 3:
+            last_graphic_control_index = gif_object.images[LAST_ELEMENT].index_graphic_control_ex
+            last_graphic_control_disposal = gif_object.graphic_control_extensions[last_graphic_control_index].disposal
+            if last_graphic_control_disposal == 3:
                 gif_object.images.append(gif_object.images[PENULTIMATE])
+            elif last_graphic_control_disposal == 2:
+                print("has disposal 2, need to check if correct")
+                background_frame(gif_object)
```
Comment (noamTzuber) [inline decoder.py:64]: we cant add with no example for test..pls find an example(maybe u can create one)

**Review by noamTzuber — CHANGES_REQUESTED** [review]

File: decoder.py (hunk 1 above)
Comment (NoamCohen48) [reply decoder.py:105]: [from what i have seen](http://www.theimage.com/animation/pages/disposal2.html), only the frame part is need to be returned to background

File: decoder.py (hunk 2 above)
Comment (NoamCohen48) [reply decoder.py:64]: i will try
also, for some reason some other changes from other branches made their way to this branch, i will recreate the branch and will pr again.

**Pull request closed without merging**